macro/private/config/yalang-yl620a.g

```diff
@@ -0,0 +1,197 @@
+    var startAddr = { var.configItem[0] }
+    var values = { var.configItem[1] }
+
+    ; Create display string for logging
+    var displayStr = { "ArborCtl: Yalang YL620-A - Writing batch " ^ (iterations + 1) ^ " - Address " ^ var.startAddr ^ ": " }
+    echo { "A: " ^ var.displayStr }
+
+    ; Build value string for display purposes only
+    var valueStr = ""
+    echo { "B: " ^ var.values }
+    while { iterations < #var.values }
+        set var.valueStr = { var.valueStr ^ var.values[iterations] ^ (iterations < #var.values - 1 ? ", " : "") }
+
+    set var.displayStr = { var.displayStr ^ var.valueStr }
+    echo { "C: " ^ var.displayStr }
+
+    ; Write batch of parameters using Modbus command - pass values vector directly to B parameter
+    while { iterations < #var.values }
+        echo { "ArborCtl: Yalang YL620-A - Writing parameter " ^ (iterations + 1) ^ ": " ^ var.values[iterations] }
+        M260.1 P{param.C} A{param.A} F6 R{var.startAddr + iterations} B{var.values[iterations]}
+        G4 P{var.waitTime}
+
+    ; Verify values were set correctly by reading them back
+    var allCorrect = true
+
+    ; Read back the parameters to verify they were set correctly
+    ; We must read these one-by-one as reading multiple bytes seems to return
+    ; incorrect values for certain registers.
+    while { iterations < #var.values }
+        M261.1 P{param.C} A{param.A} F3 R{var.startAddr + iterations} B1 V"readValue"
+        G4 P{var.waitTime}
+        if { var.readValue == null || #var.readValue != 1 }
+            echo { "ArborCtl: Yalang YL620-A - Readback failed for parameter " ^ (iterations + 1) ^ ": expected " ^ var.values[iterations] }
+            set var.allCorrect = false
+            break
+
+        if { var.readValue[0] != var.values[iterations] }
+            echo { "ArborCtl: Yalang YL620-A - Verification failed for parameter " ^ (iterations + 1) ^ ": expected " ^ var.values[iterations] ^ ", got " ^ var.readValue }
+            set var.allCorrect = false
+            break
+
+    ; Store verification result
+    set var.statusVector[iterations] = { var.allCorrect }
+
+    echo { "ArborCtl: Yalang YL620-A - Batch " ^ (iterations + 1) ^ " result: " ^ (var.allCorrect ? "Success" : "Failed") }
+
+; Report configuration results
+var successCount = 0
+while { iterations < #var.statusVector }
+    if { var.statusVector[iterations] == true }
+        set var.successCount = { var.successCount + 1 }
+
+if { var.successCount == 0}
+    M291 P{"VFD Configuration <b>failed</b>.<br/>No config batches were set successfully."} R"ArborCtl: Yalang YL620-A" S0 T5
+if { var.successCount < #global.yl620aConfigParams }
+    echo { "ArborCtl: Yalang YL620-A - Warning: Some parameters could not be set. Check VFD communication." }
+else
+    M291 P{"VFD Configuration <b>successful</b>.<br/>" ^ var.successCount ^ " of " ^ #global.yl620aConfigParams ^ " config batches set successfully."} R"ArborCtl: Yalang YL620-A" S0 T5
```

Comment:
Minor but no need to count `global.yl620aConfigParams` again in this line - it will always match `var.successCount`.